### ISSUE TITLE
rpc.statd failure when mounting NFS shares (IPX)

### ISSUE THREAD

**Issue by crawford** [coreos/bugs#575]
<a href="https://github.com/atleberg"><img src="https://avatars.githubusercontent.com/u/3176788?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [atleberg](https://github.com/atleberg)**
_Friday Jun 13, 2014 at 11:05 GMT_
_Originally opened as https://github.com/coreos/coreos-overlay/issues/684_

----

I'm trying to mount NFS shares, using cloud-config, which fails since rpc.statd can't open /etc/netconfig. I'm using the lates PXE image from the alpha channel and cloud-config for the configuration. With this setup the /etc/netconfig doesn't exist so I'm not surprised.

Touching the file manually or even filling it with a known to work gentoo configuration does not seem to help.

Original error message below follows.

core@stripe001 ~ $ journalctl -u fileserver.mount
-- Logs begin at Fri 2014-06-13 09:04:04 UTC, end at Fri 2014-06-13 09:10:36 UTC. --
Jun 13 09:04:14 stripe001 systemd[1]: Mounting /fileserver...
Jun 13 09:04:14 stripe001 rpc.statd[3044]: Version 1.2.9 starting
Jun 13 09:04:14 stripe001 rpc.statd[3044]: Flags: TI-RPC
Jun 13 09:04:14 stripe001 rpc.statd[3044]: rpc: failed to open /etc/netconfig
Jun 13 09:04:24 stripe001 mount[3022]: mount.nfs: rpc.statd is not running but is required for remote locking.
Jun 13 09:04:24 stripe001 mount[3022]: mount.nfs: Either use '-o nolock' to keep locks local, or start statd.
Jun 13 09:04:24 stripe001 mount[3022]: mount.nfs: an incorrect mount option was specified
Jun 13 09:04:24 stripe001 systemd[1]: fileserver.mount mount process exited, code=exited status=32
Jun 13 09:04:24 stripe001 systemd[1]: Failed to mount /fileserver.
Jun 13 09:04:24 stripe001 systemd[1]: Unit fileserver.mount entered failed state.

**Comment by crawford** [comment]
<a href="https://github.com/kelseyhightower"><img src="https://avatars.githubusercontent.com/u/1123322?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [kelseyhightower](https://github.com/kelseyhightower)**
_Monday Oct 13, 2014 at 20:13 GMT_

----

@atleberg  Sorry it's taken so long to get to this, but are you still have this issue with the current CoreOS releases?

**Comment by crawford** [comment]
<a href="https://github.com/atleberg"><img src="https://avatars.githubusercontent.com/u/3176788?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [atleberg](https://github.com/atleberg)**
_Wednesday Oct 15, 2014 at 13:10 GMT_

----

@kelseyhightower Did a quick test today using the latest image and it has the same issue unfortunately.

Before I try to mount the share I've tried to start the rpc-statd.service, but it fails as well:

journalctl -u rpc-statd.service
-- Logs begin at Wed 2014-10-15 12:54:25 UTC, end at Wed 2014-10-15 13:07:01 UTC. --
Oct 15 12:54:32 stripe001 systemd[1]: Dependency failed for NFSv2/3 Network Status Monitor Daemon.

To be perfectly clear; I'm trying to connect to a NFS3 share here. NFS4 isn't really an option with our current setup.

**Comment by crawford** [comment]
<a href="https://github.com/jonboulle"><img src="https://avatars.githubusercontent.com/u/2601015?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [jonboulle](https://github.com/jonboulle)**
_Wednesday Oct 15, 2014 at 18:59 GMT_

----

@atleberg hmm, if you `sudo mkdir -p /etc/conf.d && sudo touch /etc/conf.d/nfs` and then try to start `rpc-statd.service`, do things work?

**Comment by crawford** [comment]
<a href="https://github.com/atleberg"><img src="https://avatars.githubusercontent.com/u/3176788?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [atleberg](https://github.com/atleberg)**
_Friday Oct 17, 2014 at 13:24 GMT_

----

@jonboulle I tried as you suggested, but since I'm booting the machine over PXE I can't create anything inside /etc. I've set up a local disk as local storage for persistent configuration (as detailed in the documentation), but try to avoid having to install anything.

I'm guessing there's a way around this, but I'm not sure where to start.

**Comment by crawford** [comment]
<a href="https://github.com/marineam"><img src="https://avatars.githubusercontent.com/u/153640?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [marineam](https://github.com/marineam)**
_Friday Oct 17, 2014 at 20:51 GMT_

----

I will check if Gentoo has fixed the systemd units to not refer to conf.d any more, if so we just need to update our copy of the packages. If not we will need to modify them to work sanely on CoreOS after /etc has been reset. PXE images don't carry some extra stuff that the disk images do so /etc is always the bare minimum.

**Comment by crawford** [comment]
<a href="https://github.com/atleberg"><img src="https://avatars.githubusercontent.com/u/3176788?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [atleberg](https://github.com/atleberg)**
_Wednesday Oct 22, 2014 at 13:26 GMT_

----

Excellent, thanks!

Even though I'm running an PXE image; shouldn't I be able to write to /etc if I've set up local storage correctly?

**Comment by crawford** [comment]
<a href="https://github.com/ZimboBoyd"><img src="https://avatars.githubusercontent.com/u/7256778?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ZimboBoyd](https://github.com/ZimboBoyd)**
_Monday Feb 23, 2015 at 19:29 GMT_

----

I can confirm that on a PXE booted coreos 598.0.0 there is no /etc/netconfig

If i create with
```
  - path: /etc/netconfig
    permissions: 0644
    content: |
      udp        tpi_clts      v     inet     udp     -       -
      tcp        tpi_cots_ord  v     inet     tcp     -       -
      udp6       tpi_clts      v     inet6    udp     -       -
      tcp6       tpi_cots_ord  v     inet6    tcp     -       -
      rawip      tpi_raw       -     inet      -      -       -
      local      tpi_cots_ord  -     loopback  -      -       -
      unix       tpi_cots_ord  -     loopback  -      -       -
```


the file

NFS3 mounts start to work

**Comment by crawford** [comment]
<a href="https://github.com/AntonioMeireles"><img src="https://avatars.githubusercontent.com/u/743527?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [AntonioMeireles](https://github.com/AntonioMeireles)**
_Friday Mar 13, 2015 at 09:48 GMT_

----

afaict, fwiw, this is fixed in recent CoreOS alphas.

**Comment by crawford** [comment]
<a href="https://github.com/mischief"><img src="https://avatars.githubusercontent.com/u/1487996?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mischief](https://github.com/mischief)**
_Friday Mar 13, 2015 at 20:28 GMT_

----

/etc/netconfig should be present in pxe images now. also, /etc/conf.d is gone and the systemd units from the nfs ebuild no longer refer to them.

**Comment by crawford** [comment]
<a href="https://github.com/marineam"><img src="https://avatars.githubusercontent.com/u/153640?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [marineam](https://github.com/marineam)**
_Friday Mar 13, 2015 at 21:39 GMT_

----

@mischief can this be closed then?

**Comment by crawford** [comment]
<a href="https://github.com/mischief"><img src="https://avatars.githubusercontent.com/u/1487996?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mischief](https://github.com/mischief)**
_Monday Mar 16, 2015 at 18:09 GMT_

----

yes, should be solved since coreos/coreos-overlay#1130
